src/lib.rs

```diff
@@ -118,27 +121,37 @@ A: Sized + ApplicationBase<gfx_device_gl::Resources, gfx_device_gl::CommandBuffe
     let builder = glutin::WindowBuilder::from_winit_builder(wb)
                                         .with_gl(gl_version)
                                         .with_vsync();
-    let (window, mut device, factory, main_color, main_depth) =
+    let (window, mut device, mut factory, main_color, main_depth) =
         gfx_window_glutin::init::<ColorFormat, DepthFormat>(builder);
-    let (width, height) = window.get_inner_size_points().unwrap();
+    let (orig_width, orig_height) = window.get_inner_size_points().unwrap();
     let shade_lang = device.get_info().shading_language;
 
-    let mut app = A::new(factory, Init {
-        backend: if shade_lang.is_embedded {
-            shade::Backend::GlslEs(shade_lang)
-        } else {
-            shade::Backend::Glsl(shade_lang)
-        },
+    let backend = if shade_lang.is_embedded {
+        shade::Backend::GlslEs(shade_lang)
+    } else {
+        shade::Backend::Glsl(shade_lang)
+    }; 
+    let mut app = A::new(&mut factory, backend, WindowTargets {
         color: main_color,
         depth: main_depth,
-        aspect_ratio: width as f32 / height as f32,
+        aspect_ratio: orig_width as f32 / orig_height as f32,
     });
 
     let mut harness = Harness::new();
     loop {
         for event in window.poll_events() {
-            if !app.on(event) {
-                return
+            match event {
+                winit::Event::Closed => return,
+                winit::Event::KeyboardInput(winit::ElementState::Pressed, _, key) if key == A::get_exit_key() => return,
+                winit::Event::Resized(width, height) => if width != orig_width || height != orig_height {
```

Comment:
If you resize the window (this event fires as expected), and then resize again back to the original dimensions then this event will not fire even though it should.